forte/data/data_store.py

```diff
@@ -226,6 +230,217 @@ def __init__(
+                        # annotation-like
+                        self._DataStore__tid_ref_dict[
+                            e[constants.TID_INDEX]
+                        ] = e
+                    else:
+                        # non-annotation-like
+                        type_name = e[constants.ENTRY_TYPE_INDEX]
+                        if type_name in reset_index:
+                            reset_index[type_name] += 1
+                        else:
+                            reset_index[type_name] = 1
+                        self._DataStore__tid_idx_dict[type_name] = [
+                            type_name,
+                            reset_index[type_name],
+                        ]
+
+    @classmethod
+    def deserialize(
+        cls,
+        data_source: str,
+        serialize_method: str = "json",
+        check_attribute: bool = True,
+        suppress_warning: bool = True,
+        accept_none: bool = True,
+    ) -> "DataStore":
+        """
+        Deserialize a `DataStore` from serialized data in `data_source`.
+
+        Args:
+            data_source: The path storing data source.
+            serialize_method: The method used to serialize the data, this
+                should be the same as how serialization is done. The current
+                option is `json`.
+            check_attribute: Boolean value indicating whether users want to
+                check compatibility of attributes. Only applicable when the
+                data being serialized is done with `save_attribute` set to
+                True in BaseStore.serialize. If true, it will compare fields
+                of the serialized object and the current `DataStore` class.
+                If there are fields that have different orders in the current
+                class and the serialized object, it switches the order of
+                fields to match the current class.
+                If there are fields that appear in the current class, but not in
+                the serialized object, it handles those fields with
+                `accept_none`.
+                If there are fields that appear in the serialized object, but
+                not in the current class, it drops those fields.
+            suppress_warning: Boolean value indicating whether users want to
+                see warnings when it checks attributes. Only applicable when
+                `check_attribute` is set to True. If true, it will
+                log warnings when there are mismatched fields.
+            accept_none: Boolean value indicating whether users want to
+                fill fields that appear in the current class, but not in
+                the serialized object with none. Only applicable when
+                `check_attribute` is set to True. If false, it will raise
+                an `ValueError` if there are any contradictions in fields.
+
+        Returns:
+            An data store object deserialized from the string.
```

Comment:
Document the exception that can be thrown by this function

Reply:
yes